src/flow-timeout.c

```diff
@@ -401,6 +401,7 @@ static inline void FlowForceReassemblyForHash(void)
                 RemoveFromHash(f, prev_f);
                 f->flow_end_flags |= FLOW_END_FLAG_SHUTDOWN;
                 FlowForceReassemblyForFlow(f);
```

Comment:
Should we not put the unlock in the FlowForceReassemblyForFlow ? As it is only queuing the flow in another thread this should be systematic.

Reply:
`FlowForceReassemblyForFlow` accesses a field in the flow, so technically it should do so under lock. I don't think it would matter much either way in this case.